apps/api/src/server/applications/documents/documents.service.js

```diff
@@ -0,0 +1,42 @@
+import { EventType } from '@pins/event-client';
+import { eventClient } from '../../infrastructure/event-client';
+import { NSIP_DOCUMENT } from '../../infrastructure/topics';
+import { buildNsipDocumentPayload } from '../application/documents/document';
+import * as documentRepository from '../../repositories/document.repository.js';
+import * as documentVersionRepository from '../../repositories/document-metadata.repository.js';
+import { nextStatusInDocumentStateMachine } from '../application/application.service';
+
+/**
+ * @param {string} guid
+ * @param {string} newStatus
+ */
+export const updateStatus = async (guid, newStatus) => {
+	const document = await documentRepository.getByDocumentGUID(guid);
+
+	const currentDocumentStatus = document?.documentVersion[0]?.publishedStatus;
+
+	/** @type {string} */
+	// @ts-ignore
+	const status = nextStatusInDocumentStateMachine(currentDocumentStatus, newStatus);
```

Comment:
call `nextStatus` ?

Reply:
It's a good call, but it's passed as a named parameter on the next line and the parameter is called 'status', so just keeping consistent with the repository parameter (but also so I don't have to rename it when passing)